dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/BrowserAPI.java

```diff
@@ -254,256 +248,111 @@ public WfData(Contentlet contentlet, List<Integer> permissions, User user, boole
 
 		if (!onlyFiles) {
 
-			// Getting the html pages directly under the parent folder or host
-			List<IHTMLPage> pages = new ArrayList<IHTMLPage>();
-			try {
-				if (parent != null) {//For folders
-					if(!showWorking) {
-						pages.addAll(APILocator.getHTMLPageAssetAPI().getLiveHTMLPages(parent, user, false));
-					}
-					else {
-						pages.addAll(APILocator.getHTMLPageAssetAPI().getWorkingHTMLPages(parent, user, false));
-					}
+	        // Getting the links directly under the parent folder or host
+	        List<Link> links = new ArrayList<Link>();
 
-					if(showArchived) {
-						pages.addAll(APILocator.getHTMLPageAssetAPI().getDeletedHTMLPages(parent, user, false));
-					}
-				} else {//For hosts
-                    if ( !showWorking ) {
-                        pages.addAll( APILocator.getHTMLPageAssetAPI().getLiveHTMLPages( host, user, false ) );
-                    } else {
-                        pages.addAll( APILocator.getHTMLPageAssetAPI().getWorkingHTMLPages( host, user, false ) );
-                    }
-
-                    if ( showArchived ) {
-                        pages.addAll( APILocator.getHTMLPageAssetAPI().getDeletedHTMLPages( host, user, false ) );
-                    }
-				}
-			} catch (Exception e1) {
-				Logger.error(this, "Could not load HTMLPages : ", e1);
-			}
+            if (parent != null) {
+                if (showWorking) {
+                    links.addAll(folderAPI.getLinks(parent, true, false, user, false));
+                } else {
+                    links.addAll(folderAPI.getLiveLinks(parent, user, false));
+                }
+                if (showArchived)
+                    links.addAll(folderAPI.getLinks(parent, true, showArchived, user, false));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [This statement should have braces](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5337824)